### ISSUE TITLE
E_NOINTERFACE returned for ICoreWebView2Experimental with Edge 89.0.713.0 canary

### ISSUE THREAD

**Issue by dd8** [MicrosoftEdge/WebView2Feedback#691]
**Description**
E_NOINTERFACE is returned when querying ICoreWebView2 for ICoreWebView2Experimental when using Edge 89.0.713.0 canary

**Version**
SDK: 1.0.674-prerelease
Runtime: Edge 89.0.713.0 canary
Framework: Win32, etc.
OS:  Win10 version 1909

**Repro Steps**
Using code example from https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2experimental?view=webview2-1.0.674-prerelease&preserve-view=true#add_webresourceresponsereceived modified to use Microsoft::WRL::ComPtr instead of wil::com_ptr

Expected behavior: QueryInterface returns S_OK and interface pointer
Actual behavior: QueryInterface returns E_NOINTERFACE and null interface pointer

Version reported by ICoreWebView2Environment get_BrowserVersionString is Edge 89.0.713.0 canary (when WEBVIEW2_RELEASE_CHANNEL_PREFERENCE=1)

**Installed components**
Microsoft Edge 87.0.664.47
Microsoft Edge Canary 89.0.713.0
Microsoft Edge Update 1.3..139.59
Microsoft Edge WebView2 Runtime 87.0.664.47
WEBVIEW2_RELEASE_CHANNEL_PREFERENCE=1

**Additional context**
This looks similar to https://github.com/MicrosoftEdge/WebView2Feedback/issues/630

[AB#30900659](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/30900659)

**Comment by champnic** [comment]
@dd8 Thanks for the bug report! We think we understand the issue, and have a bug on our backlog.

As a workaround, could you try using `CreateCoreWebView2EnvironmentWithOptions` and pass in an environment options object? You can see an example here:
https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2environmentoptions?view=webview2-1.0.674-prerelease

We think the bug is that in the default case a default environment options object is not getting created, and so failing to include the SDK info needed for the runtime to use an older, compatible CoreWebView2 object.

**Comment by dd8** [comment]
That's exactly it. When calling CreateCoreWebView2EnvironmentWithOptions with a NULL environmentOptions, querying for ICoreWebView2Experimental on the web view returns E_NOINTERFACE:

```
CreateCoreWebView2EnvironmentWithOptions( nullptr, userDataFolder, nullptr, .... );
```

ICoreWebView2Experimental is returned as expected if we change the create call to:

```
auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
CreateCoreWebView2EnvironmentWithOptions( nullptr, userDataFolder, options.Get(), .... );
```

**Comment by champnic** [comment]
Thanks for verifying, glad the workaround works! We'll get this fixed for the next SDK release.

**Comment by champnic** [comment]
This should be fixed in the latest 1.0.781-prerelease SDK package. Thanks!
